internal/graph/graph.go

```diff
@@ -117,13 +117,13 @@ func (g *ConnectedObjectGraph) findIngresses(target *openfgapb.RelationReference
 	key := tuple.ToObjectRelationString(target.GetType(), target.GetRelation())
 	if _, ok := visited[key]; ok {
 		// We've already visited the target so no need to do so again.
-		return nil, nil
+		return make([]*RelationshipIngress, 0), nil
```

Comment:
in the graph unit tests we do [`require.ElementsMatch(t, test.expected, ingresses)`](https://github.com/openfga/openfga/blob/main/internal/graph/graph_test.go#L854). When those tests break the output of the diff was gigantic so i temporarily switched to `cmp.Diff`, which was very nitpicky about the fact that a nil response and a zero array response are not the same.

Reply:
Let's pass compare options to `cmp.Diff` then so that we can keep this contract the same. The semantics of returning a nil slice if a non-nil error is returned is idiomatic Go.